backend/app/routes/routes.py

```
@@ -77,9 +78,20 @@ async def run_pipelines(request: URlRequest):
+        article_context = (request.article_context or "").strip()
+
+        results = search_pinecone(query)
+        answer = ask_llm(query, results, article_context)
+        logger.info("Chat answer generated successfully.")
+
+        return {"answer": answer}
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.exception(f"Chat request failed: {e}")
+        raise HTTPException(status_code=500, detail="Failed to generate chat response.")
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)